test/frontend/Readme.md

```diff
@@ -0,0 +1,140 @@
+
+Open a python shell and run the following code:
+
+    from selenium import webdriver
+    cdriver = webdriver.Chrome()
+    fdriver = webdriver.Firefox()
+
+If any problems arise at this point, check whether the driver that was downloaded supports the
+actually installed corresponding browser.
+
+From the same source download the Selenium standalone server, something 
+along the lines of `selenium-server-standalone-[version].jar`, 
+and have it available.
+
+To allow parallel and cross browser testing, and even giving the possibility of testing
+with different operating systems and on different machines, we next set up selenium grid.
+In order to do this, we need the selenium standalone server, which is already included as a java jar 
+in the "selenium" directory. Be sure to check out newer versions from time to time on the same page as above.
+A hub is set up  accessible via port 4444 by default, where nodes with different browsers can be added.
+
+The hub can be started by opening a terminal, changing to the selenium folder and then typing in the following command
+(replace with you current version of the jar):
+
+```
+java -jar selenium-server-standalone-3.141.59.jar -role hub
+```
+
+Leave the terminal running, open another terminal, navigate to the directory and add a node to the hub:
+
+```
+java -jar selenium-server-standalone-3.141.59.jar -role node
+```
+
+If you now visit [http://localhost:4444/grid/console](http://localhost:4444/grid/console/) you can see the node 
+registered and which browsers are available on each of them.
+You can set the port, the browser running on the node and various other parameters. 
+It is also possible now to register nodes from different machines, given you know the IP of the hub, by expanding the 
+command above to
+
+```
+java -jar selenium-server-standalone-3.141.59.jar -role node -hub http://$IP of hub$:4444/grid/register
+```
+ 
+Using just your local machine, you can also add more nodes via the same command,
+speeding up the run time of the tests, as they will get distributed over these nodes by pytest.
+For more helpful information about selenium grid, check out [the docs](https://www.seleniumhq.org/docs/07_selenium_grid.jsp) 
+and this [article on medium.com](https://medium.com/@arnonaxelrod/running-cross-browser-selenium-tests-in-parallel-using-selenium-grid-and-docker-containers-9ee293b86cfd)
+
+After this preparation, open yet another terminal. Before starting the tests, be sure to create a file containing 
+session cookies:
+
+```
+python load_cookies.py
+```
+
+Now we can run the tests via the `pytest` or alternatively the `python -m pytest` command. Without any arguments,
+all tests in the directory will be executed.
+Specific tests can always be run by adding the file name to this command.
+Using the `pytest-xdist` module, parallel testing is possible, decreasing the overall run time of the tests.
```

Comment:
Maybe mention that pytest-xdist should be installed separately (it's not included in pytest).

Reply:
Good point, will do.